File: MFDLabs.Grid.Bot/Commands/CloseExpiredJobs.cs
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;
using System.Threading.Tasks;

namespace MFDLabs.Grid.Bot.Commands
{
    internal class CloseExpiredJobs : IStateSpecificCommandHandler
    {
        public string CommandName => "Close All Expired Jobs";

        public string CommandDescription => "Closes all the grid server's expired jobs.";

        public string[] CommandAliases => new string[] { "caej", "closeallexpiredjobs" };

        public bool Internal => true;

        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            if (!await message.RejectIfNotAdminAsync()) return;

            await SoapUtility.Singleton.CloseExpiredJobsAsync();
            await message.ReplyAsync("Successfully closed all expired grid server jobs.");
        }
    }
}

File: Source/Lib/Commands/CloseExpiredJobs.ts
import { Message } from 'discord.js';
import { AdminUtil } from '../Administration/AdminUtil';
import { SOAPUtil } from '../RCCService/SOAPUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';

class CloseExpiredJobs implements StateSpecificCommandHandler {
	public CommandName = 'Close All Expired Jobs';
	public CommandDescription = 'Closes all expired jobs';
	public Internal = true;
	public Command = ['caej', 'closeallexpiredjobs', 'closeallexpiredgames'];
	public Callback = async (messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		await SOAPUtil.CloseExpiredJobs();
		return messageRoot.reply(`Successfully closed all expired jobs.`);
	};
}

export = new CloseExpiredJobs();
